### DISCUSSION TITLE
How to parse a time stamp with a length of 16 (microseconds)

### DISCUSSION THREAD

**Discussion by h7ml** [moment/moment#6425] [Q&A] [answered]
I need to parse a 16 bit microsecond timestamp
Now i have this format with microsecond 
```1651107155498222```
I checked the relevant [documents ](https://momentjs.com/docs/#/displaying/format/)and found that the moment only supports parsing to milliseconds
How can i add microseconds ?

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for asking! Moment and JavaScript `Date` have millisecond precision. I'd divide the microsecond timestamp by `1000` before constructing the Moment and preserve the remainder separately if it matters. With large values, `BigInt` avoids losing integer precision: `moment(Number(micros / 1000n))`.

Note, [Temporal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal) supports microseconds, and nanoseconds too!
